gumrya-web/src/test/java/gumrya/service/UserServiceTest.java

```diff
@@ -0,0 +1,130 @@
+package gumrya.service;
+
+import gumrya.dto.UserRequest;
+import gumrya.dto.UserResponse;
+import gumrya.exception.DuplicateException;
+import gumrya.exception.NotFoundException;
+import gumrya.user.domain.model.User;
+import gumrya.user.domain.repository.UserRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+
+@ExtendWith(MockitoExtension.class)
+@DisplayName("User 비지니스 흐름 테스트")
+class UserServiceTest {
+
+    @InjectMocks
+    private UserService userService;
+
+    @Mock
+    private UserRepository userRepository;
+
+    private User user;
+    private UserRequest userRequest;
+
+    @BeforeEach
+    void setup() {
+        this.userRequest = new UserRequest(1L, "ryan@gmail.com", "12342n1k3l", "검프", "나는 검프다", "www.url.com",
+                "github" +
+                        ".com/younghoonkwon");
+        this.user = new User(userRequest.getId(), userRequest.getEmail(), userRequest.getPassword(),
+                userRequest.getNickname(),
+                userRequest.getIntroduction(), userRequest.getProfileUrl(), userRequest.getGithubUrl());
+    }
+
+    @DisplayName("사용자 저장 - 성공")
+    @Test
+    void save() {
+        //given
+        given(userRepository.save(any(User.class))).willReturn(user);
+        UserResponse userResponse = new UserResponse(user);
+        //when
+        UserResponse saveUser = userService.save(userRequest);
+        //then
+        assertThat(saveUser).isEqualTo(userResponse);
+    }
+
+    @DisplayName("사용자 저장 - 실패")
+    @Test
+    void saveFail() {
+        //given
+        given(userRepository.existsByEmail(any(String.class))).willReturn(true);
+        //when
+        //then
+        assertThatThrownBy(() -> userService.save(userRequest)).isInstanceOf(DuplicateException.class);
```

Comment:
```suggestion
        assertThatThrownBy(() -> userService.save(userRequest))
        .isInstanceOf(DuplicateException.class);
```
줄바꿈을 하는 것을 컨벤션으로 잡아도 괜찮을거같아요(검증 대상이 많아질 경우 줄바꿈이 필요하기 때문)